sgtool/file.go

```diff
@@ -189,11 +191,19 @@ func WithSkipIfFileExists(filepath string) Opt {
 	}
 }
 
-func downloadBinary(ctx context.Context, url string) (io.ReadCloser, func(), error) {
+func WithHTTPHeader(key, value string) Opt {
```

Comment:
This API seems a bit limiting (what about multiple values for the same header?) - why not do `http.Header` as argument?

Reply:
As [mentioned](https://github.com/einride/sage/pull/182#discussion_r805593612) I don't have a strong preference. Adding multiple values for the same header will still work as long as we do `f.httpHeader.Add(key, value)` rather than use `.Set()`

Reply:
hehe that is what he used originally :D

Reply:
I agree that for this specific scenario however that is not very obvious simply looking at the API though